pkg/core/blockchain.go

```diff
@@ -490,65 +477,52 @@ func (bc *Blockchain) addHeaders(verify bool, headers ...*block.Header) (err err
 		}
 	}
 
-	bc.headersOp <- func(headerList *HeaderHashList) {
-		oldlen := headerList.Len()
-		for _, h := range headers {
-			if int(h.Index-1) >= headerList.Len() {
-				err = fmt.Errorf(
-					"height of received header %d is higher then the current header %d",
-					h.Index, headerList.Len(),
-				)
-				return
-			}
-			if int(h.Index) < headerList.Len() {
-				continue
-			}
-			if err = bc.processHeader(h, batch, headerList); err != nil {
-				return
-			}
+	buf := io.NewBufBinWriter()
+	bc.headerHashesLock.Lock()
+	defer bc.headerHashesLock.Unlock()
+	oldlen := len(bc.headerHashes)
+	var lastHeader *block.Header
+	for _, h := range headers {
+		if int(h.Index-1) >= len(bc.headerHashes) {
+			return fmt.Errorf("height of received header %d is higher than the current header %d", h.Index, len(bc.headerHashes))
```

Comment:
Perhaps now we can get rid of this named `err` return value?

Reply:
Fixed.